demos/common/cpp/pipelines/src/requests_pool.cpp

```diff
@@ -14,28 +14,30 @@
 // limitations under the License.
 */
 
+#include <vector>
+#include <openvino/openvino.hpp>
 #include "pipelines/requests_pool.h"
 
-RequestsPool::RequestsPool(InferenceEngine::ExecutableNetwork& execNetwork, unsigned int size) :
+RequestsPool::RequestsPool(ov::runtime::CompiledModel& compiledModel, unsigned int size) :
     numRequestsInUse(0) {
     for (unsigned int infReqId = 0; infReqId < size; ++infReqId) {
-        requests.emplace(std::make_shared<InferenceEngine::InferRequest>(execNetwork.CreateInferRequest()), false);
+        requests.emplace(std::make_shared<ov::runtime::InferRequest>(compiledModel.create_infer_request()), false);
     }
 }
 
 RequestsPool::~RequestsPool() {
     // Setting empty callback to free resources allocated for previously assigned lambdas
     for (auto& pair : requests) {
-        pair.first->SetCompletionCallback([]{});
+        pair.first->set_callback([](const std::exception_ptr& e) {});
     }
 }
 
-InferenceEngine::InferRequest::Ptr RequestsPool::getIdleRequest() {
+RequestsPool::InferRequestPtr RequestsPool::getIdleRequest() {
     std::lock_guard<std::mutex> lock(mtx);
 
-    const auto& it = std::find_if(requests.begin(), requests.end(), [](std::pair<const InferenceEngine::InferRequest::Ptr, bool>& x) {return !x.second; });
+    const auto& it = std::find_if(requests.begin(), requests.end(), [](std::pair<const InferRequestPtr, bool>& x) {return !x.second; });
     if (it == requests.end()) {
-        return InferenceEngine::InferRequest::Ptr();
+        return std::make_shared<ov::runtime::InferRequest>(ov::runtime::InferRequest());
```

Comment:
You should use just `InferRequest()`. ireq is just a pair of shared_ptrs

Reply:
Done.